include/logging/log.h

```diff
@@ -275,6 +274,17 @@ int log_printk(const char *fmt, va_list ap);
  *	   a buffer from the pool (see CONFIG_LOG_STRDUP_MAX_STRING).
  */
 char *log_strdup(const char *str);
+#else
+static inline void log_printk(const char *fmt, va_list ap)
+{
+	vprintk(fmt, ap);
+}
+
+static inline char *log_strdup(const char *str)
+{
+	return (char *)str;
```

Comment:
what would happen if someone expects this function to duplicate the string because they want to re-use the same string buffer?

```
const char *bt_addr_str_real(const bt_addr_t *addr)
{
	static char str[BT_ADDR_STR_LEN];
	bt_addr_to_str(addr, str, sizeof(str));
	return str;
}
LOG_INFO("%s connected to %s",
         log_strdup(bt_addr_str_real(local_addr)),
         log_strdup(bt_addr_str_real(remote_addr)));
```

Reply:
that won't work. This calls for adding custom format specifiers there are so far two options on the table:
- linux style, extending printk to support custom formatters (eg `%pba` means bluetooth address)
- PR #17367 where i proposed to extend printk to support user specified formatters (implemented in the module and not in printk). In that case this call would look something like:
```
LOG_INF("%s connected to %s", log_bt_addr_le_t_dup(local_addr), log_bt_addr_le_t_dup(remote_addr))
```

Reply:
> . This calls for adding custom format specifiers 

No.
We are not adding custom format specifiers.
You are adding a dependency on having format strings available at runtime, and we don't want that for binary logging. You are also breaking the ability for the compiler to check format codes which we don't want either.
I've already made my case clear in #17367 this is NOT happening.

>  there are so far two options on the table:

No.
There are more than your two proposed options.
You can log individual struct members with already existing format codes.
You can write custom functions to log bluetooth addresses, which in turn logs individual struct members with already existing format codes.
You can just dump the whole object to the log stream and get the specific data you want out of it later.

Reply:
No, I'm not talking about custom format specificatiors, I'm talking about string duplication. Something that is currently not working as expected between the normal logging, and the CONFIG_LOG_IMMEDIATE, and CONFIG_LOG_MINIMAL, where the string in `log_strdup` is not duplicated, but simply returned.